src/React.js

```diff
@@ -72,12 +79,7 @@ function createClassWithDerivedState(classCtr) {
   };
 }
 
-var componentImpl = createClass(React.Component);
-exports.componentImpl = componentImpl;
 exports.componentWithDerivedStateImpl = createClassWithDerivedState(componentImpl);
```

Comment:
These `*WithDerivedState` functions were moved after `createClassWithDerivedState()` to satisfy eslint's rules about not using functions before they are defined.